### ISSUE TITLE
Batch Get All stops processing if it encounters an empty set

### ISSUE THREAD

**Issue by Dave-Dohmeier** [guregu/dynamo#161]
When calling Batch Get All if an empty result set is found it stops processing there and returns an error not found without checking any more keys.

**Working**
```
Adding test key "1"
Adding test key "200"
Checking keys 1-200
Found: [1 200]
```
**Failing**
```
Adding test key "200"
Checking keys 1-200
Found: []
```

Key 200 is not found because after the first 100 keys are checked with no matches the Get operation stops running and returns an `ErrNotFound`. This should only be returned if nothing was found from all the keys.

**Comment by guregu** [comment]
Are you running the batch get with `Consistent(true)`? Can you set AWS's config log level to LogDebugWithHTTPBody and make sure there are actually unprocessed keys? Or post the code?

```go
&aws.Config{
	Region: aws.String(region),
	LogLevel: aws.LogLevel(aws.LogDebugWithHTTPBody),
}
```

**Comment by Dave-Dohmeier** [comment]
**Output**
```
Results from batch get: [{Value:1} {Value:200}]
Error returned from batch getdynamo: no item found
Results from batch get: []
```

**Code**
```
package main

import (
	"fmt"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/credentials"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/guregu/dynamo"
)

type Entry struct {
	Value string `dynamo:"value"`
}

func main() {

	tableName := "blocklist"

	db := dynamo.New(session.New(), &aws.Config{
		Region: aws.String("us-east-1"),
		// LogLevel:    aws.LogLevel(aws.LogDebugWithHTTPBody),
		Credentials: credentials.NewStaticCredentials("foo", "bar", "baz"),
		Endpoint:    aws.String("http://dynamodb.default.svc.cluster.local"),
	})
	table := db.Table(tableName)

	entry := Entry{Value: "1"}
	table.Put(entry).Run()
	entry = Entry{Value: "200"}
	table.Put(entry).Run()

	keysToCheck := []dynamo.Keyed{}
	for i := 1; i <= 200; i++ {
		keysToCheck = append(keysToCheck, dynamo.Keys{fmt.Sprintf("%d", i)})
	}

	results := []Entry{}
	err := table.Batch("value").Get(keysToCheck...).All(dynamo.AWSEncoding(&results))
	if err != nil {
		fmt.Printf("Error returned from batch get%+v\n", err)
	}
	fmt.Printf("Results from batch get: %+v\n", results)

	table.Delete("value", "1").Run()

	results = []Entry{}
	err = table.Batch("value").Get(keysToCheck...).All(dynamo.AWSEncoding(&results))
	if err != nil {
		fmt.Printf("Error returned from batch get%+v\n", err)
	}
	fmt.Printf("Results from batch get: %+v\n", results)
}
```

I'm somewhat sure it's coming from here where any batch with 0 results just terminates looking for any more:
https://github.com/guregu/dynamo/blob/master/batchget.go#L270

**Comment by guregu** [comment]
Thanks for the code, I'll see if I can reproduce.

**Comment by guregu** [comment]
@Dave-Dohmeier 
Thanks a bunch, I was able to reproduce and fix it. I just released v1.10.4 which includes the fix.

The cause was indeed that part you linked. For empty sets whose index was divisible by 100, it wasn't checking the remaining keys properly, only looking at the response instead of the total count. Sorry about that.
